pkg/beholder/auth.go

```diff
@@ -0,0 +1,23 @@
+package beholder
+
+import "fmt"
+
+// authHeaderKey is the name of the header that the node authenticator will use to send the auth token
+var authHeaderKey = "X-Beholder-Node-Auth-Token"
+
+// authHeaderVersion is the version of the auth header format
+var authHeaderVersion = "1"
+
+// BuildAuthHeaders creates the auth header value to be included on requests.
+// The current format for the header is:
+//
+// <version>:<public_key_hex>:<signature_hex>
+//
+// where the byte value of <public_key_hex> is what's being signed
+func BuildAuthHeaders(signer func([]byte) []byte, pubKey []byte) map[string]string {
+	messageBytes := pubKey
+	signature := signer(messageBytes)
```

Comment:
moved to accepting an ed25519 key instead of a signer [with this feedback](https://github.com/smartcontractkit/chainlink-common/pull/877#discussion_r1833222924)